client/control/components/ActiveFilters.js

```diff
@@ -0,0 +1,88 @@
+import React, { PropTypes as pt } from 'react';
+
+/**
+ * Section above RecipeList's table that displays
+ * active filter options and settings, and allows
+ * users to reset all/individual filters.
+ */
+
+export default class ActiveFilters extends React.Component {
+  static propTypes = {
+    selectedFilters: pt.array.isRequired,
+    onResetFilters: pt.func.isRequired,
+    onFilterSelect: pt.func.isRequired,
+    className: pt.string,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+
+    this.handlerCache = {};
+  }
+
+  handleFilterSelect(group, option) {
+    const cacheKey = group.value + option.value;
+
+    // check if an existing event handler exists
+    if (!this.handlerCache[cacheKey]) {
+      // if not, create it with the group and option given
+      this.handlerCache[cacheKey] = () =>
+        this.props.onFilterSelect(group, option);
+    }
+
+    // return the handling function
+    return this.handlerCache[cacheKey];
+  }
+
+  /**
+   * Render
+   */
+  render() {
+    const {
+      className,
+      selectedFilters,
+      onResetFilters,
+    } = this.props;
+
+    // optional className prop
+    const displayedClass = `active-filters ${className || ''}`;
```

Comment:
I think you can do this a little cleaner with [classnames](https://www.npmjs.com/package/classnames), which we already have:

```js
import cx from 'classnames';

let className = undefined;
cx('active-filters', className);  # 'active-filters'
className = 'some-class';
cx('active-filters', className);  # 'active-filters some-class';
```